internal/lsp/server.go

```diff
@@ -686,6 +709,18 @@ func (s *Server) handleInitialized(ctx context.Context, params *lsproto.Initiali
 		NpmExecutor: s,
 		ParseCache:  s.parseCache,
 	})
+
+	// request userPreferences if not provided at initialization
+	if s.initializeParams.InitializationOptions == nil {
+		userPreferences, err := s.RequestConfiguration(ctx)
+		if err != nil {
+			return err
+		}
+		s.session.Configure(userPreferences)
+	} else {
+		// !!! handle userPreferences from initialization
```

Comment:
When will we ever have these?

Reply:
requires changes in the extension, which I haven't done yet

Reply:
theoretically, it would be nice to have userpreferences (and other preferences/config info) passed over in the actual `initialization` request through [`initializeParams.initializationOptions`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initializeParams), so we don't have to wait for a `configuration` request to come back. It doesn't seem to be included by default, so we'd have to fix the extension to give extra info in `initializationOptions`

Reply:
I suppose so, but I've never seen a server which did that, honestly... It just means copying the same "get all of the options" code into the client another time, and then it gets copied into other clients, to save one roundtrip.